src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SnsService.kt

```diff
@@ -0,0 +1,132 @@
+
+  fun sendVisitBookedEvent(visit: VisitDto) {
+    log.debug("Sending visit booked event")
+    publishToDomainEventsTopic(
+      HMPPSDomainEvent(
+        eventType = EVENT_PRISON_VISIT_BOOKED,
+        version = EVENT_PRISON_VISIT_VERSION,
+        description = EVENT_PRISON_VISIT_BOOKED_DESC,
+        occurredAt = Instant.now().toString(),
+        prisonerId = visit.prisonerId,
+        additionalInformation = AdditionalInformation(
+          reference = visit.reference,
+        )
+      )
+    )
+  }
+
+  fun sendVisitCancelledEvent(visit: VisitDto) {
+    log.debug("Sending visit cancelled event")
+    publishToDomainEventsTopic(
+      HMPPSDomainEvent(
+        eventType = EVENT_PRISON_VISIT_CANCELLED,
+        version = EVENT_PRISON_VISIT_VERSION,
+        description = EVENT_PRISON_VISIT_CANCELLED_DESC,
+        occurredAt = Instant.now().toString(),
+        prisonerId = visit.prisonerId,
+        additionalInformation = AdditionalInformation(
+          reference = visit.reference
+        )
+      )
+    )
+  }
+
+  private fun publishToDomainEventsTopic(payload: HMPPSDomainEvent) {
+    log.debug("Event ${payload.eventType} for id ${payload.additionalInformation.reference}")
+
+    try {
+      domaineventsTopicClient.publish(
+        PublishRequest(domaineventsTopic.arn, objectMapper.writeValueAsString(payload))
+          .withMessageAttributes(
+            mapOf(
+              "eventType" to MessageAttributeValue().withDataType("String").withStringValue(payload.eventType)
+            )
+          ).also { log.info("Published event $payload to outbound topic") }
+      )
+    } catch (e: Throwable) {
+      // throw PublishEventException("Failed to publish Event $payload.eventType to $TOPIC_ID", e)
+
+      // Note: Silently fail until VB-671 is implemented
+      log.debug("Failed to publish Event $payload.eventType to $TOPIC_ID", e)
+    }
+  }
+
+  companion object {
+    const val TOPIC_ID = "domainevents"
+    const val EVENT_ZONE_ID = "Europe/London"
+    const val EVENT_PRISON_VISIT_VERSION = 1
+    const val EVENT_PRISON_VISIT_BOOKED = "prison-visit.booked"
+    const val EVENT_PRISON_VISIT_BOOKED_DESC = "Prison Visit Booked"
+    const val EVENT_PRISON_VISIT_CANCELLED = "prison-visit.cancelled"
+    const val EVENT_PRISON_VISIT_CANCELLED_DESC = "Prison Visit Cancelled"
+
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+}
+
+data class AdditionalInformation(
+  val reference: String,
+)
+
+@Suppress("unused")
+data class HMPPSDomainEvent(
+  val eventType: String,
+  val version: Int,
+  val description: String,
+  val occurredAt: String,
+  val prisonerId: String,
+  val additionalInformation: AdditionalInformation,
+) {
+  constructor(
+    eventType: String,
+    description: String,
+    occurredAt: Instant,
+    prisonerId: String,
+    additionalInformation: AdditionalInformation
+  ) : this(
+    eventType,
+    1,
+    description,
+    occurredAt.toOffsetDateFormat(),
```

Comment:
Nice, time zones...

Reply:
This need a good test something i wanted a second pair of eyes on when it comes to roundtrip serialisation and object mapping constructors.